src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/InMemoryAssemblyLoader.cs

```diff
@@ -23,9 +21,15 @@ public static class InMemoryAssemblyLoader
         /// <param name="assemblyPath">The path to the assembly (as a pointer to a UTF-16 C string).</param>
         public static unsafe void LoadInMemoryAssembly(IntPtr moduleHandle, IntPtr assemblyPath)
         {
+            string? assemblyPathString = Marshal.PtrToStringUni(assemblyPath);
+            if (assemblyPathString == null)
+            {
+                throw new ArgumentOutOfRangeException(nameof(assemblyPath));
+            }
```

Comment:
@sdmaclea, as these are new types in .NET Core 3, I added in some argument validation rather than letting things throw NullReferenceException on bad inputs.